src/main/java/uk/gov/pay/ledger/agreement/resource/AgreementResource.java

```diff
@@ -91,4 +89,22 @@ public Agreement get(@Parameter(description = "The unique external id for the ag
                 .map(Agreement::from)
                 .orElseThrow(() -> new WebApplicationException(Response.Status.NOT_FOUND));
     }
-}
\ No newline at end of file
+
+    @Path("/{agreementExternalId}/event")
+    @GET
+    @Timed
+    @Operation(
+            summary = "Find agreement events by agreement ID",
+            responses = {
+                    @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = Agreement.class))),
```

Comment:
schema is `Agreement.class` but schema returned is `List<EventDto>`

Reply:
Oops, good spot.  I've changed it to `AgreementEventsResponse.class`, by analogy with a similar endpoint in `TransactionResource`.